app/controllers/merchant/invoices_controller.rb

```diff
@@ -1,5 +1,11 @@
 class Merchant::InvoicesController < ApplicationController 
   def index 
-    @invoices = Invoice.all
+    @merchant = Merchant.find(params[:merchant_id])
+    @invoices = @merchant.invoices.merchant_invoices
```

Comment:
I'm fairly certain we can just use `.distinct` at the end of this line, rather than creating a whole method that just does exactly that. This shouldn't violate MCV principles, but we can check if we end up making that change.